Signal/src/ViewControllers/ConversationView/ConversationLayoutInfo.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+
+@objc
+public class ConversationLayoutInfo: NSObject {
+
+    private let thread: TSThread
+
+    private let isRTL: Bool
+
+    // The width of the collection view.
+    @objc public var viewWidth: CGFloat = 0 {
+        didSet {
+            SwiftAssertIsOnMainThread(#function)
+
+            updateProperties()
+        }
+    }
+
+    @objc public let contentMarginTop: CGFloat = 10
+    @objc public let contentMarginBottom: CGFloat = 10
+
+    @objc public var gutterLeading: CGFloat = 0
```

Comment:
Are we really mutating instances of ConversationLayoutInfo? Or can these all be readonly?

Reply:
* Some of the future properties (added in later PRs) are mutable, e.g. to reflect changes to dynamic type size.
* I'm trying to build this logic to make future work to support orientation changes easy, where these views will automatically update to reflect changes to the root view size.